test/util/collect.js

```diff
@@ -0,0 +1,61 @@
+"use strict";
+
+const wp = require("../../lib/webidl2");
+const pth = require("path");
+const fs = require("fs");
+const jdp = require("jsondiffpatch");
+
+/**
+ * Collects test items from the specified directory
+ * @param {string} base
+ * @param {boolean} [expectError]
+ */
+function* collect(base, expectError) {
```

Comment:
Let's make `expectError` not be a boolean. Maybe make it an object, so to avoid the boolean trap?